apps/ledger-live-desktop/src/renderer/components/RecoverRestore/index.tsx

```diff
@@ -1,34 +1,100 @@
+import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
+import getVersion from "@ledgerhq/live-common/hw/getVersion";
+import { first } from "rxjs/operators";
+import { from } from "rxjs";
+import {
+  OnboardingState,
+  extractOnboardingState,
+} from "@ledgerhq/live-common/hw/extractOnboardingState";
+import { FirmwareInfo } from "@ledgerhq/types-live";
+import { renderError } from "../DeviceAction/rendering";
 
 const RecoverRestore = () => {
   const { t } = useTranslation();
   const history = useHistory();
   const currentDevice = useSelector(getCurrentDevice);
+  const [state, setState] = useState<OnboardingState>();
+  const [error, setError] = useState<Error>();
   const { setDeviceModelId } = useContext(OnboardingContext);
 
+  // check if device is seeded when selected
   useEffect(() => {
-    switch (currentDevice?.modelId) {
-      case DeviceModelId.nanoX:
-        setDeviceModelId(currentDevice.modelId);
-        history.push(`/onboarding/${UseCase.recover}/${ScreenId.pairMyNano}`);
-        break;
-      case DeviceModelId.stax:
-        history.push(`/onboarding/sync/${currentDevice.modelId}`);
-        break;
-      default:
-        break;
+    if (currentDevice) {
+      const requestObservable = withDevice(currentDevice.deviceId)(t => from(getVersion(t))).pipe(
+        first(),
+      );
+
+      const sub = requestObservable.subscribe({
+        next: (firmware: FirmwareInfo) => {
+          try {
+            setState(extractOnboardingState(firmware.flags));
+          } catch (error: unknown) {
+            if (error instanceof Error) {
+              setError(error);
+            }
+          }
+        },
+        error: (error: Error) => {
+          setError(error);
+        },
+      });
+
+      return () => {
+        sub.unsubscribe();
+      };
+    }
+  }, [currentDevice]);
+
+  useEffect(() => {
+    if (state && !state.isOnboarded) {
+      switch (currentDevice?.modelId) {
+        case DeviceModelId.nanoX:
+          setDeviceModelId(currentDevice.modelId);
+          history.push(`/onboarding/${UseCase.recover}/${ScreenId.pairMyNano}`);
+          break;
+        case DeviceModelId.stax:
+          history.push(`/onboarding/sync/${currentDevice.modelId}`);
+          break;
+        default:
+          break;
+      }
     }
-  }, [currentDevice?.modelId, history, setDeviceModelId]);
+  }, [currentDevice?.modelId, history, setDeviceModelId, state]);
+
+  if (error) {
+    return renderError({
+      t,
+      error,
+      device: currentDevice,
+    });
+  }
+
+  if (state?.isOnboarded) {
+    return (
+      <Flex width="100%" height="100%" position="relative">
+        <Flex position="relative" height="100%" width="100%" flexDirection="column">
+          <OnboardingNavHeader onClickPrevious={() => history.push("/onboarding/select-device")} />
+          {renderError({
+            t,
+            error: new DeviceOnboarded("Your device is already setup with a Secret Recover Phrase"),
```

Comment:
All user-facing errors should define a translation key, rather than define the message in the constructor, [here](https://github.com/LedgerHQ/ledger-live/blob/178d65887d7e6d67ddda8b83181c9fc1a3b0a13b/apps/ledger-live-desktop/static/i18n/en/app.json#L5026) for LLD ([here](https://github.com/LedgerHQ/ledger-live/blob/178d65887d7e6d67ddda8b83181c9fc1a3b0a13b/apps/ledger-live-mobile/src/locales/en/common.json) for LLM)